### DISCUSSION TITLE
Restarting OpenVPN service

### DISCUSSION THREAD

**Discussion by ckooistra** [jaredhendrickson13/pfsense-api#233] [Q&A]
Hi there I have a specific use case for which I installed this package.  I would like to be able to reboot my openvpn client service.  I see that I can restart all services but is there a way to pass parameters to target specific services?  Sorry I have been looking around in the documentation but not seeing what I am looking for.  Thanks for your help!

**Comment by jaredhendrickson13** [comment]
Hey!

The /api/v1/services/restart endpoint has the `service` property that you can use to specify a single service to restart. To only restart OpenVPN, your request body to this endpoint should be: `{"service": "openvpn"}`

![Screen Shot 2022-05-24 at 8 35 42 AM](https://user-images.githubusercontent.com/50180862/170061866-36558096-c6be-4b90-9ad3-067059aec49e.png)

Hope this helps.

Thanks!

**Comment by ckooistra** [comment]
Hi I really appreciate the reply.  In fact I was wondering if it is
possible to start only a specific client connection of the vpn service not
the whole service itself!

**Comment by ckooistra** [comment]
in fact I always get this message back

**Comment by ckooistra** [comment]
[image: image.png]

**Comment by jaredhendrickson13** [comment]
Currently, no. There is not a supported method of restarting OpenVPN clients. There is an open feature request for this though (#136). Feel free to follow that issue for updates on future progress.

**Comment by cappiz** [comment]
I don't seem to be able to restart the openvpn service. In my case I got two openvpn server.
The restart api does not throw any error, but in my logs I can't see that the server has been reloaded.

Any suggestions?
